### ISSUE TITLE
Opacity of DragPreview

### ISSUE THREAD

**Issue by EmptyCrown** [react-dnd/react-dnd#668]
Hi, is it possible to change the opacity of the drag preview? Whenever, I drag something the preview is somewhat transparent. How do we change this?

**Comment by edgesoft** [comment]
You could implement your own with DragLayer

**Comment by EmptyCrown** [comment]
I see, thanks. However, it seems that with a custom DragLayer you have to specify your own DragPreview? How would I still obtain the "snapshot" of the dragged element and just adjust its opacity?

**Comment by edgesoft** [comment]
@EmptyCrown Sorry. It's seems it's not possible

https://github.com/react-dnd/react-dnd/issues/549

**Comment by theTechie** [comment]
@edgesoft - you can by using custom styles.

Check this example - https://github.com/react-dnd/react-dnd/blob/master/examples/02%20Drag%20Around/Custom%20Drag%20Layer/CustomDragLayer.js#L17

you can pass in any styles you want.

**Comment by edgesoft** [comment]
@theTechie Think the question was "How would I still obtain the "snapshot" of the dragged element and just adjust its opacity?"

**Comment by theTechie** [comment]
The "snapshot"  we see without using any DragLayer is the browser snapshot, we don't have control on it. That's the reason we have DragLayer to customize.

Let's say, you want to render components `<A />, <B /> and <C />`; you can have a DragLayer similar to the one referred above and ensure you render the same element `<A />, <B /> or <C />` by using the types and props passed on to the DragLayer.

```
switch (type) {
      case ItemTypes.A:
        return <A {...props} />);
      case ItemTypes.B:
        return <B {...props} />);
      default:
        return null;
}
```

You will get a custom drag preview (but it will actually be the same component you had initially rendered and you get control to customize it)

PS: if you look at the example referred above, it's exactly what is being done. `<BoxDragPreview />` is just `<Box />` wrapped in another <div /> with custom styling.

**Comment by anilreddykatta** [comment]
@EmptyCrown Does above comment address your issue? If yes, please can you close this issue?

**Comment by PavelMikhadziuk** [comment]
@theTechie Hi. I also used `react-dnd` and` DragLayer` for customize a preview. I added `transform: rotate (7deg)` for my `CustomDragLayer` (similar `BoxDragPreview.js` from example). But I have one trouble. Element which wrapping in `connectDragSource` (DragSourceConnector which is in component wrapped in `connectDropTarget`) do not rotate. How am I going to solve this problem? Thank you. And excuse me for I did not create a separate issue.
